tests/test_two_view_estimator_cacher.py

```diff
@@ -0,0 +1,121 @@
+"""Unit tests for two-view estimator cacher."""
+
+import unittest
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import numpy as np
+from gtsam import Cal3Bundler
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.two_view_estimator_cacher import TwoViewEstimatorCacher
+
+
+ROOT_PATH = Path(__file__).resolve().parent.parent
+
+_DUMMY_OUTPUT = (None, None, None, None, None, None)
+
+
+class TestTwoViewEstimatorCacher(unittest.TestCase):
+    def setUp(self) -> None:
+        # Generate 20 random keypoints, and assume H = W = 1000.
+        coordinates_i1 = np.random.randint(low=0, high=1000, size=(20, 2))
+        coordinates_i2 = np.random.randint(low=0, high=1000, size=(20, 2))
+
+        self.keypoints_i1 = Keypoints(coordinates_i1)
+        self.keypoints_i2 = Keypoints(coordinates_i2)
+        # Horizontally stack two (5,1) arrays to form (20,2).
+        self.corr_idxs = np.hstack([np.arange(5).reshape(-1, 1)] * 2)
+
+        fx, k1, k2, u0, v0 = 583.1175, 0, 0, 507, 380
+        calibration = Cal3Bundler(fx, k1, k2, u0, v0)
+        self.camera_intrinsics_i1 = calibration
+        self.camera_intrinsics_i2 = calibration
+
+        self.dummy_output = _DUMMY_OUTPUT
+
+    @patch("gtsfm.utils.cache.generate_hash_for_numpy_array", return_value="numpy_key")
+    @patch("gtsfm.utils.io.read_from_bz2_file", return_value=None)
+    @patch("gtsfm.utils.io.write_to_bz2_file")
+    def test_cache_miss(
+        self, write_mock: MagicMock, read_mock: MagicMock, generate_hash_for_numpy_array_mock: MagicMock
+    ) -> None:
+        """Test the scenario of cache miss for the TwoViewEstimator."""
+
+        # Mock the underlying two-view estimator which is used on cache miss.
+        underlying_estimator_mock = MagicMock()
```

Comment:
I think the `underlying` word is unnecessary..

Reply:
Acknowledged, I think it's helpful : - )